pyiron/atomistics/job/structurecontainer.py

```diff
@@ -2,62 +2,160 @@
+    """
+
+    __version__ = "0.1.0"
 
-class StructureContainer(AtomisticGenericJob):
     def __init__(self, project, job_name):
-        super(StructureContainer, self).__init__(project, job_name)
-        self.__version__ = "0.1"
-        self.__name__ = "StructureContainer"
+        super().__init__(project, job_name)
+        self.__structures = InputList(table_name = "structures")
         self.server.run_mode.interactive = True
 
     @property
-    def structure(self):
-        return self._structure
+    def structures(self):
+        """
+        :class:`.InputList`: list of structures
+        with meta data; each item in this list has an 'atoms' key for the
+        atomic structures and then as many additional keys as passed to
+        :meth:`~.StructureContainer.append()`
+        """
+        return self.__structures
 
-    @structure.setter
-    def structure(self, structure):
-        if isinstance(structure, AtomisticGenericJob):
+    @staticmethod
+    def __to_structure(structure_or_job):
+        """
+        If :class:`~.AtomisticGenericJob` try to get most recent structure, if
+        :class:`~.Atoms` return as is, throw an error otherwise.  Wrap
+        structures in dict under the key "atoms".
+        """
+        if isinstance(structure_or_job, AtomisticGenericJob):
             if structure.structure:
-                self._structure = structure.get_structure(iteration_step=-1)
-                self.parent_id = structure.job_id
+                return {
+                        "atoms": structure_or_job.get_structure(iteration_step=-1),
+                        "jobid": structure_or_job.job_id
+                }
             else:
-                raise ValueError("The job does not contain any structure to import.")
-        elif isinstance(structure, Atoms):
-            self._structure = structure
+                raise ValueError(
+                        "The job does not contain any structure to import."
+                )
+        elif isinstance(structure_or_job, Atoms):
+            return {"atoms": structure_or_job}
         else:
             raise TypeError(
-                "You can only append a structure object or an GenericJob object."
+                "You can only use a structure object or an "
+                "AtomisticGenericJob object."
             )
-        if self.status.initialized:
-            self.run()
 
-    def append(self, structure_to_append):
-        self.structure = structure_to_append
+    @property
+    def structure(self):
+        """
+        :class:`~.Atoms`: first (or only) structure set in the container
+        """
+        return self.structures[0].atoms
+
+    @structure.setter
+    def structure(self, structure):
+        item = self.__to_structure(structure_or_job)
+        if len(self.structures) >= 1:
+            self.structures[0] = item
+        else:
+            self.structures.append(item)
+
+    def append(self, structure_or_job, **kwargs):
```

Comment:
I am a bit reluctant to change the signature of append, can not we require a dictionary to begin with? And when the user tries to append only a single structure create the dictionary internally?

Reply:
I thought
```python
job.append(..., foo = bar)
```
feels more concise that
```python
job.append({'atoms': ..., 'foo': bar})
```
This also requires the user to know about the correct key to store the structure under, which I feel is an implementation detail.  I could rename it to something else, though.

Reply:
To me:
```
job.append(..., foo = bar)
```
does not feel like python. Still we might also be able to store this description as tags for the atomistic structure or like ASE is using an "info" dictionary, we should be able to use that as well https://wiki.fysik.dtu.dk/ase/_modules/ase/atoms.html

Reply:
Forgot about `Atoms.info`! Will use that.